libs/editor/WordPressEditor/src/main/java/org/wordpress/android/editor/gutenberg/GutenbergDialogFragment.kt

```diff
@@ -0,0 +1,139 @@
+    private lateinit var mPositiveButtonLabel: CharSequence
+    private var mTitle: CharSequence? = null
+    private var mNegativeButtonLabel: CharSequence? = null
+    private var mMediaId: Int = 0
+    private var dismissedByPositiveButton = false
+    private var dismissedByNegativeButton = false
+
+    interface GutenbergDialogPositiveClickInterface {
+        fun onGutenbergDialogPositiveClicked(instanceTag: String, mediaId: Int)
+    }
+
+    interface GutenbergDialogNegativeClickInterface {
+        fun onGutenbergDialogNegativeClicked(instanceTag: String)
+    }
+
+    interface GutenbergDialogOnDismissByOutsideTouchInterface {
+        fun onDismissByOutsideTouch(instanceTag: String)
+    }
+
+    fun initialize(
+        tag: String,
+        title: CharSequence?,
+        message: CharSequence,
+        positiveButtonLabel: CharSequence,
+        negativeButtonLabel: CharSequence? = null,
+        mediaId: Int
+    ) {
+        mTag = tag
+        mTitle = title
+        mMessage = message
+        mPositiveButtonLabel = positiveButtonLabel
+        mNegativeButtonLabel = negativeButtonLabel
+        mMediaId = mediaId
+    }
+
+    override fun onCreate(savedInstanceState: Bundle?) {
+        super.onCreate(savedInstanceState)
+        this.isCancelable = true
+        val theme = 0
+        setStyle(STYLE_NORMAL, theme)
+
+        if (savedInstanceState != null) {
+            mTag = requireNotNull(savedInstanceState.getString(STATE_KEY_TAG))
+            mTitle = savedInstanceState.getCharSequence(STATE_KEY_TITLE)
+            mMessage = requireNotNull(savedInstanceState.getCharSequence(STATE_KEY_MESSAGE))
+            mPositiveButtonLabel = requireNotNull(savedInstanceState.getCharSequence(STATE_KEY_POSITIVE_BUTTON_LABEL))
+            mNegativeButtonLabel = savedInstanceState.getCharSequence(STATE_KEY_NEGATIVE_BUTTON_LABEL)
+            mMediaId = savedInstanceState.getInt(STATE_KEY_MEDIA_ID)
+        }
+    }
+
+    override fun onSaveInstanceState(outState: Bundle) {
+        outState.putString(STATE_KEY_TAG, mTag)
+        outState.putCharSequence(STATE_KEY_TITLE, mTitle)
+        outState.putCharSequence(STATE_KEY_MESSAGE, mMessage)
+        outState.putCharSequence(STATE_KEY_POSITIVE_BUTTON_LABEL, mPositiveButtonLabel)
+        outState.putCharSequence(STATE_KEY_NEGATIVE_BUTTON_LABEL, mNegativeButtonLabel)
+        outState.putInt(STATE_KEY_MEDIA_ID, mMediaId)
+        super.onSaveInstanceState(outState)
+    }
+
+    override fun onCreateDialog(savedInstanceState: Bundle?): Dialog {
+        val builder = MaterialAlertDialogBuilder(requireActivity())
+        builder.setMessage(mMessage)
+
+        mTitle?.let {
+            builder.setTitle(mTitle)
+        }
+
+        mPositiveButtonLabel?.let {
+            builder.setPositiveButton(mPositiveButtonLabel) { _, _ ->
+                dismissedByPositiveButton = true
+                val parentFragment: Fragment? = parentFragment
```

Comment:
Hey Siobhan, I know this patter of having a same-name local const picking up the value from a getter (`getParentFragment()` in this case) [is used in similar places](https://github.com/wordpress-mobile/WordPress-Android/blob/develop/WordPress/src/main/java/org/wordpress/android/ui/posts/BasicFragmentDialog.kt#L83-L86), but makes me a bit uncomfortable when reading the code. Trusted, in AndroidStudio with its syntax highlighting it's kinda easy to see the difference between the two, but it's no super intuitive otherwise.

I was thinking, would you be interested in using the following to do the same null-check and casting directly, without the temporary val?

```kotlin
               (parentFragment as? GutenbergDialogPositiveClickInterface)?.let {
                    it.onGutenbergDialogPositiveClicked(mTag, mMediaId);
                }
```

Reply:
Thank you for this, it led me down a road of learning [a bit more about Kotlin](https://discuss.kotlinlang.org/t/let-vs-if-not-null/3542/11) and I've gone ahead to implement your suggestion. :)

Reply:
Ah, nice! While at it, wdyt about doing the same treatment to [this null-check and cast case in `onDismiss()`](https://github.com/wordpress-mobile/WordPress-Android/pull/14503/files#diff-8889a43bfa80f8522889ae2f69695ac5eda5b29189bb29c9915a9c4825dae6e7R116-R122) too?

Reply:
Ah, good point! I've gone ahead to do that in https://github.com/wordpress-mobile/WordPress-Android/pull/14503/commits/7924341c7b430ec48d6fbb56d2f4d6a8866491c9, as part of that commit I also changed the reference to `activity` to `parentFragment`, to match the way the other interfaces are implemented. Let me know if you spot any issues with that!

Reply:
Thanks! I [left a comment on the commit](https://github.com/wordpress-mobile/WordPress-Android/commit/7924341c7b430ec48d6fbb56d2f4d6a8866491c9#r51332384).